GameOfLife/GameOfLife/Game/BoardGame.cs

```diff
@@ -0,0 +1,44 @@
+﻿
+namespace GameOfLife
+{
+    /// <summary>
+    /// Implementation for games that use Board interface.
+    /// </summary>
+    public class BoardGame : Game<IBoard>
+    {
+        /// <summary>
+        /// Board of the game.
+        /// </summary>
+        private IBoard _gameBoard { get; set; }
+
+        /// <summary>
+        /// Constructor for the game.
+        /// </summary>
+        public BoardGame(IBoard gameBoard) => _gameBoard = gameBoard;
+
+        /// <inheritdoc/>
+        public override void Play()
+        {
+            while (State == GameState.Playing)
+            {
+                Action keyActions = () => Panel.DisplayKeyMenu(
```

Comment:
This comment is related to comment for .DisplayKeyMenu() method.
Implementation is too complex. Your method DisplayKeyMenu() should be just displaying menu as its name states.
Possible simple solution would be:
-change DisplayKeyMenu() so that it just displays menu,
- there should be other method to read user's input, 
- it should be followed by switch statement which processes user's selection,
- two last items could be wrapped in if(Console.KeyAvailable()){}